app/javascript/mastodon/reducers/timelines.js

```diff
@@ -31,10 +31,10 @@ const initialTimeline = ImmutableMap({
 });
 
 const normalizeTimeline = (state, timeline, statuses, next) => {
-  const ids       = ImmutableList(statuses.map(status => status.get('id')));
+  const oldIds    = state.getIn([timeline, 'items'], ImmutableList());
+  const ids       = ImmutableList(statuses.map(status => status.get('id'))).filter(newId => !oldIds.includes(newId));
```

Comment:
I think most cases of this issue is when the status (A) arrives via streaming while timeline refreshing (Bs) due to column changing. So existing status A would be newer than fetched items. However, this implementation places fetched statuses (`ids`) before newer existing status A.

I think it's better if we can remove duplicated statuses from `oldIds` and prefer fetched `ids` (ordered correctly), but not must be I think.